### PR TITLE
JerryScript api improvemts accordign to jerry_api_value_t struct.

### PR DESCRIPTION
**Pull request by ktorpi**
JerryScript-DCO-1.0-Signed-off-by: István Kádár ikadar@inf.u-szeged.hu

### CODE REVIEW

**Comment by egavrin** [comment]
Please, fix missing comments and incorrect `*` formatting in this patch.

**Comment by ktorpi** [comment]
I've added missing comments in jerry.cpp and fixed pointer formatting.

**Comment by egavrin** [comment]
Please merge commits into single one and LGTM.

File: jerry-core/jerry-api.h [hunk 1]
```diff
@@ -146,6 +146,149 @@ typedef void (*jerry_object_free_callback_t) (const uintptr_t native_p);
...
+ */
+extern EXTERN_C jerry_api_object_t *
+jerry_api_get_object_value (const jerry_api_value_t *value_p);
+
+/**
+ * Creates and returns a jerry_api_value_t with type
+ * JERRY_API_DATA_TYPE_VOID.
+ */
+extern EXTERN_C jerry_api_value_t
+jerry_api_create_void_value ();
```
Comment (galpeter) [inline jerry-core/jerry-api.h]: AFAIK: our stlye "dictates" that we when there is no argument for a function we should add the `void`, for example this should be:
```
jerry_api_create_void_value (void);
```

File: jerry-core/jerry.cpp [hunk 2]
```diff
@@ -124,6 +124,242 @@ jerry_make_api_unavailable (void)
 } /* jerry_make_api_unavailable */
 
 /**
+ * Returns whether the given jerry_api_value_t is void.
+ */
+bool
+jerry_api_value_is_void (const jerry_api_value_t *value_p)
+{
+  return value_p->type == JERRY_API_DATA_TYPE_VOID;
+}
```
Comment (galpeter) [inline jerry-core/jerry.cpp]: Missing argument comments and function end comments (as per style), this should be:
```
bool
method_name (int arg) /** << my super comment on the argument */
{
...
} /* method_name */
```

**Comment by ktorpi** [comment]
I've fixed void parameters, added comments on parameters and function endings and merged commits into a single one.

**Comment by galpeter** [comment]
lgtm

**Comment by ruben-ayrapetyan** [comment]
Merged (636ded489f8ea896763dc91ad91c045ff2918957)

**Pull request closed without merging**